client/web/compose/src/components/Admin/Page/Builder/Selector.vue

```diff
@@ -33,30 +33,74 @@
       </b-col>
 
       <hr
-        v-if="existingBlocks.length"
+        v-if="existingLayoutBlocks.length || selectableNamespaceGlobalBlocks"
         class="w-100"
       >
 
       <b-col
-        v-if="existingBlocks.length"
+        v-if="selectableNamespaceGlobalBlocks.length"
         cols="12"
       >
         <b-input-group class="d-flex w-100">
           <c-input-select
-            v-model="selectedExistingBlock"
+            v-model="selectedGlobalBlock"
+            :get-option-label="getBlockLabel"
+            :get-option-key="getOptionKey"
+            :options="selectableNamespaceGlobalBlocks"
+            :reduce="b => b.blockID"
+            placeholder="Select global block from other pages"
+          />
+
+          <b-input-group-append>
+            <b-button
+              v-b-tooltip.noninteractive.hover="{ title: $t('selector.tooltip.clone.noRef'), container: '#body' }"
+              variant="extra-light"
+              :disabled="!selectedGlobalBlock"
+              class="d-flex align-items-center"
+              @click="$emit('select', fetchBlockData(selectedGlobalBlock).clone())"
```

Comment:
Since these @click are the same for both input groups, can you make them in a method, so its less code?